cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/store/DefaultMetadataStore.java

```diff
@@ -412,14 +447,34 @@ public void apply(MetadataDataset input) throws Exception {
     // Sort the results by score
     List<Map.Entry<NamespacedEntityId, Integer>> resultList = new ArrayList<>(weightedResults.entrySet());
     Collections.sort(resultList, SEARCH_RESULT_DESC_SCORE_COMPARATOR);
+    Set<NamespacedEntityId> result = new LinkedHashSet<>(resultList.size());
+    for (Map.Entry<NamespacedEntityId, Integer> entry : resultList) {
+      result.add(entry.getKey());
+    }
+    return result;
+  }
 
-    // Fetch metadata for entities in the result list
-    // Note: since the fetch is happening in a different transaction, the metadata for entities may have been
-    // removed. It is okay not to have metadata for some results in case this happens.
-    Map<NamespacedEntityId, Metadata> systemMetadata = fetchMetadata(weightedResults.keySet(), MetadataScope.SYSTEM);
-    Map<NamespacedEntityId, Metadata> userMetadata = fetchMetadata(weightedResults.keySet(), MetadataScope.USER);
+  private SortInfo getSortInfo(@Nullable String sort) throws BadRequestException {
```

Comment:
I think it would be helpful is this was a static method in SortInfo. Would make it easily unit-testable and allow validation to be done in other places if needed, like directly in the handler.